### PR TITLE
RA-543 tobivogel trim whitespaces on setting PersonName name attributes

### PR DESCRIPTION
**Pull request by tobivogel**

### CODE REVIEW

**Comment by dkayiwa** [comment]
@tobivogel why do you think that the validator is not the right place? I somehow feel that the pattern of silently changing user data without notifying them can be confusing. But i could be wrong depending on what your thoughts are. :)

**Comment by lluismf** [comment]
Imho the best approach to do it is with an hibernare custom type. But it should be a project level decision not in just one entity.

File: api/src/test/java/org/openmrs/PersonNameTest.java [hunk 1]
```diff
@@ -383,7 +437,92 @@ public void getFullName_shouldReturnShortIfPersonNameFormatIsNull() throws Excep
...
+
+		assertThat(pn.getFamilyNameSuffix(), is("und Taxis"));
+	}
+
+	/**
+	 * @see PersonName#setGivenName(String)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces", method = "setGivenName(String)")
+	public void setGivenName_shouldTrimLeadingAndTrailingWhitespaces() {
```
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l510)

File: api/src/test/java/org/openmrs/PersonNameTest.java [hunk 2]
```diff
@@ -383,7 +437,92 @@ public void getFullName_shouldReturnShortIfPersonNameFormatIsNull() throws Excep
...
+
+		assertThat(pn.getFamilyNamePrefix(), is("van der"));
+	}
+
+	/**
+	 * @see PersonName#setFamilyNameSuffix(String)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces", method = "setFamilyNameSuffix(String)")
+	public void setFamilyNameSuffix_shouldTrimLeadingAndTrailingWhitespaces() {
```
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l498)

File: api/src/test/java/org/openmrs/PersonNameTest.java [hunk 3]
```diff
@@ -383,7 +437,92 @@ public void getFullName_shouldReturnShortIfPersonNameFormatIsNull() throws Excep
...
+
+		assertThat(pn.getFamilyName(), is("van Dyke"));
+	}
+
+	/**
+	 * @see PersonName#setFamilyName2(String)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces", method = "setFamilyName2(String)")
+	public void setFamilyName2_shouldTrimLeadingAndTrailingWhitespaces() {
```
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l474)

File: api/src/test/java/org/openmrs/PersonNameTest.java [hunk 4]
```diff
@@ -383,7 +437,92 @@ public void getFullName_shouldReturnShortIfPersonNameFormatIsNull() throws Excep
...
+
+		assertThat(pn.getGivenName(), is("Hans Peter"));
+	}
+
+	/**
+	 * @see PersonName#setMiddleName(String)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces", method = "setMiddleName(String)")
+	public void setMiddleName_shouldTrimLeadingAndTrailingWhitespaces() {
```
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l522)

File: api/src/test/java/org/openmrs/PersonNameTest.java (hunk 1 above)
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l510)

File: api/src/test/java/org/openmrs/PersonNameTest.java [hunk 5]
```diff
@@ -383,7 +437,92 @@ public void getFullName_shouldReturnShortIfPersonNameFormatIsNull() throws Excep
...
+
+		assertThat(pn.getFamilyName2(), is("von Hohenlohe"));
+	}
+
+	/**
+	 * @see PersonName#setFamilyNamePrefix(String)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces", method = "setFamilyNamePrefix(String)")
+	public void setFamilyNamePrefix_shouldTrimLeadingAndTrailingWhitespaces() {
```
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l486)

File: api/src/test/java/org/openmrs/PersonNameTest.java [hunk 6]
```diff
@@ -383,7 +437,92 @@ public void getFullName_shouldReturnShortIfPersonNameFormatIsNull() throws Excep
...
+
+		Assert.assertEquals("Sr. Taylor Bob Mark", pn.getFullName());
+	}
+
+	/**
+	 * @see PersonName(String, String, String)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces on each name attribute", method = "PersonName")
+	public void personName_shouldTrimLeadingAndTrailingWhitespacesOnEachNameAttribute() throws Exception {
```
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l449)

File: api/src/test/java/org/openmrs/PersonNameTest.java (hunk 2 above)
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l498)

File: api/src/test/java/org/openmrs/PersonNameTest.java (hunk 3 above)
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l474)

File: api/src/test/java/org/openmrs/PersonNameTest.java [hunk 7]
```diff
@@ -78,6 +78,60 @@ public void newInstance_shouldCopyEveryPropertyOfGivenPersonName() throws Except
 	}
 
 	/**
+	 * @see PersonName#newInstance(PersonName)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces on each name attribute", method = "newInstance(PersonName)")
+	public void newInstance_shouldTrimLeadingAndTrailingWhitespacesOnEachNameAttribute() throws Exception {
```
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l85)

File: api/src/test/java/org/openmrs/PersonNameTest.java [hunk 8]
```diff
@@ -383,7 +437,92 @@ public void getFullName_shouldReturnShortIfPersonNameFormatIsNull() throws Excep
...
+		assertThat(pn.getMiddleName(), is("Robert"));
+		assertThat(pn.getFamilyName(), is("Jones"));
+	}
+
+	/**
+	 * @see PersonName#setFamilyName(String)
+	 */
+	@Test
+	@Verifies(value = "should trim leading and trailing whitespaces", method = "setFamilyName(String)")
+	public void setFamilyName_shouldTrimLeadingAndTrailingWhitespaces() {
```
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l462)

File: api/src/test/java/org/openmrs/PersonNameTest.java (hunk 4 above)
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l522)

File: api/src/test/java/org/openmrs/PersonNameTest.java (hunk 5 above)
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l486)

File: api/src/test/java/org/openmrs/PersonNameTest.java (hunk 6 above)
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l449)

File: api/src/test/java/org/openmrs/PersonNameTest.java (hunk 7 above)
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l85)

File: api/src/test/java/org/openmrs/PersonNameTest.java (hunk 8 above)
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Codacy Issue found: [Method names should not contain underscores](https://www.codacy.com/app/openmrs/openmrs-core/file/2318747982/issues/source?bid=3204099&fileBranchId=3204128#l462)

File: api/src/test/java/org/openmrs/PersonNameTest.java (hunk 1 above)
Comment (rkorytkowski) [inline api/src/test/java/org/openmrs/PersonNameTest.java]: Please ignore this kind of issues. It is a misconfigured analysis rule.

**Comment by dkayiwa** [comment]
Let us have the validator just reject names with leading or trailing white spaces instead of silently trimming user data.

**Pull request closed without merging**